src/profile-logic/call-tree.js

```diff
@@ -584,3 +586,89 @@ export function extractSamplesLikeTable(
       throw assertExhaustiveCheck(strategy);
   }
 }
+
+/**
+ * This function is extremely similar to computeCallTreeCountsAndTimings,
+ * but is specialized for converting sample counts into traced timing. Samples
+ * don't have duration information associated with them, it's mostly how long they
+ * were observed to be running. This function computes the timing the exact same
+ * way that the stack chart will display the information, so that timing information
+ * will agree. In the past, timing was computed by samplingInterval * sampleCount.
+ * This caused confusion when switching to the trace-based views when the numbers
+ * did not agree. In order to remove confusion, we can show the sample counts,
+ * plus the traced timing, which is a compromise between correctness, and consistency.
+ */
+export function computeTracedTiming(
+  samples: SamplesLikeTable,
+  { callNodeTable, stackIndexToCallNodeIndex }: CallNodeInfo,
+  interval: Milliseconds,
+  invertCallstack: boolean
+): TracedTiming | null {
+  if (samples.weightType && samples.weightType !== 'samples') {
+    // Only compute for the samples table.
+    return null;
+  }
+
+  // Compute the timing duration, which is the time between this sample and the next.
+  const weight = [];
```

Comment:
Hmm, this may add to the complexity of the implementation because the `SamplesTable` type is serializable. This particular usage will not be since it's in a selector. I'd prefer to pass on this one.

Reply:
Don't we use typed arrays in other places too? Anyway yeah, I was mostly thinking of doing it in this algorithm, and missed that we then assign the result of that to the table. I'm not sure where this has an impact, but this can definitely be researched in a follow up if we feel like this algorithm needs to be made faster.